main.py

```diff
@@ -18,15 +18,37 @@
 from flask import Flask, render_template, redirect, url_for, request, abort, Response, jsonify
 from db_operations import *
 from forms import AdoptionForm, CreateListingForm
+from listings import Listing
+
 
 app = Flask(__name__)
 app.config['SECRET_KEY'] = os.urandom(32)
 
 
-@app.route('/')
+@app.route('/', methods=['GET', 'POST'])
 def root():
-    listings = get_listings()
-    return render_template('main.html', listings=listings)
+    if request.method == 'GET':
+        listings = get_listings()
+        form = CreateListingForm()
+        if form.validate_on_submit():
+            print(form.email)
+            f = form.upload_img.data
+            print(f)
+            # TODO: Save the img file into Firebase (?)
+        return render_template('main.html', listings=listings, form=form)
+    if request.method == 'POST':
+        form_dict = request.form.to_dict()
+        new_listing = Listing(
+            form_dict['animal'],
+            form_dict['breed'],
+            form_dict['dob'],
+            form_dict['description_of_pet'],
+            'img_url',
+            form_dict['email'],
+        )
+        create_listing(new_listing)
+        return redirect(url_for('root'))
+
```

Comment:
form.validate_on_submit() checks if it is a POST request, so I think we can just move the creation of the listing to under this function and don't need the if request.method part. Cos if we check for GET method, it will never go into the if form.validate_on_submit() block.

Reply:
Changed to put under validate_on_submit() block only